rust/sdk-processor/src/steps/token_v2_processor/token_v2_extractor.rs

```diff
@@ -0,0 +1,147 @@
+use crate::utils::database::ArcDbPool;
+use aptos_indexer_processor_sdk::{
+    aptos_protos::transaction::v1::Transaction,
+    traits::{async_step::AsyncRunType, AsyncStep, NamedStep, Processable},
+    types::transaction_context::TransactionContext,
+    utils::errors::ProcessorError,
+};
+use async_trait::async_trait;
+use processor::{
+    db::common::models::{
+        token_models::{token_claims::CurrentTokenPendingClaim, tokens::TableMetadataForToken},
+        token_v2_models::{
+            v1_token_royalty::CurrentTokenRoyaltyV1,
+            v2_collections::{CollectionV2, CurrentCollectionV2},
+            v2_token_activities::TokenActivityV2,
+            v2_token_datas::{CurrentTokenDataV2, TokenDataV2},
+            v2_token_metadata::CurrentTokenV2Metadata,
+            v2_token_ownerships::{CurrentTokenOwnershipV2, TokenOwnershipV2},
+        },
+    },
+    processors::token_v2_processor::parse_v2_token,
+};
+
+/// Extracts fungible asset events, metadata, balances, and v1 supply from transactions
+pub struct TokenV2Extractor
+where
+    Self: Sized + Send + 'static,
+{
+    query_retries: u32,
+    query_retry_delay_ms: u64,
+    conn_pool: ArcDbPool,
+}
+
+impl TokenV2Extractor {
+    pub fn new(query_retries: u32, query_retry_delay_ms: u64, conn_pool: ArcDbPool) -> Self {
```

Comment:
@rtso not passing in the config here to avoid another match statement?

Reply:
yeah just copy/pasted from the old processor